sparkly/session.py

```diff
@@ -108,6 +114,29 @@ def _create_spark_context():
         self.catalog_ext = SparklyCatalog(self)
 
         attach_writer_to_dataframe()
+        SparklySession._instantiated_session = self
+
+    @classmethod
+    def get_or_create(cls):
+        """Access instantiated sparkly session.
+
+        If sparkly session has already been instantiated, return that
+        instance; if not, then instantiate one and return it. Useful
+        for lazy access to the session. Not thread-safe.
+
+        Returns:
+            SparklySession (or subclass).
+        """
+        if cls._instantiated_session is None:
```

Comment:
Maybe better to attach `_instantiated_session` to `SparklySession` instead of `cls`.  Then, if it exists already, check to make sure is is an instance of cls before returning, and throw an exception if it is not.  

if I do something like: 
```
class A(SparklySession):
    pass
mySession = A()
SparklySession.get_or_create(cls)  # should return mySession
```

Reply:
Good point, thanks! 

Didn't throw an exception if there's a class mismatch as I'm not sure how we want to use the feature in such a case - maybe when you're in a notebook you want to deliberately mix up different `SparklySession`s (i.e., you're importing code from different projects, and you know that one is a superset of the other, so you're ok with just re-using that first one). Since this sounds like a valid usecase, and since in most projects we'd have just one type anyway, I'll leave this end kinda open and we can revisit once we form a stronger opinion on it.

Reply:
I was thinking of the exception as more of a safeguard so you never call MySession.getOrCreate() and get a session that doesn't actually have all the attributes of MySession, which could be a difficult bug to track down, if the only difference is a missing dependency or option. It shouldn't make a difference in most cases, though.